### ISSUE TITLE
Unintuitive `bind` functionality

### ISSUE THREAD

**Issue by KPull** [bastion-dev/Bastion#44]
Bastion users have reported that `bind()` does not work very intuitively. In particular, the following request fails (assuming the server returns content-type `application/json`):

```java
String data = Bastion.request("Create Sushi Request", new CreateSushiRequest())
                .bind(String.class)
                .withAssertions((statusCode, response, model) -> {
                    assertThat(statusCode).isEqualTo(201);
                }).call().getModel();
```

A user would expect that all the HTTP response entity data will be available as a `String`. Right now though, since the content-type is `application/json`, Bastion creates a *single* `ObjectNode` model and cannot return a `String`.

Here's my idea (looking for feedback):

Bastion will register a number of `ResponseDecoder`s (as it does now). Each `ResponseDecoder` may return a view object (they could return nothing). These views will be constructed by parsing the HTTP response entity data into various objects. For example, the following response decoders:

* `ObjectNodeResponseDecoder` - Returns an `ObjectNode` if the HTTP response content-type is `application/json`.
* `JsonObjectMapperResponseDecoder` - If the content-type is JSON, takes the object type inside the `DecodingHints` object and attempts to parse the HTTP response into an instance of that object type.
* `YamlObjectMapperResponseDecoder` - Same as above, but for YAML.
* `ProtobufObjectMapperResponseDecoder` - Same as above, but for Protobuf binary data.
* `MapResponseDecoder` - Provides a map representation if the response is a collection of key/value pairs (eg. Form URL encoded data).
* `ImageResponseDecoder` - If the content-type is an image, this will return an `Image` object
* `StringResponseDecoder` - Provides a `String` view of the HTTP response entity regardless of the content-type.
* ... and so on.

Now, `bind()` will assert whether a view of the given type exists in the Response. This view will be provided to the `Assertions` when the assertions are executed. Furthermore, after the `call()` method, the user may:

* Call `getModel()` to retrieve the view specified in `bind()` earlier.
* Call `getView()`, providing the view object type which the user wants.
* Call `getResponse()`, which returns the HTTP response representation. The user may then call `getModel()` and `getView()` on the response object itself which will behave just like the above two methods.

All this allows us to extend for new future views simply by implementing more `ResponseDecoder`s and it also frees us from having to decide (and guess) which *single* `ResponseDecoder` the user wants us to execute to get the bound model.

**Comment by KPull** [comment]
@tyrone-grech might want to provide some feedback on the above.

**Comment by tyrone-grech** [comment]
I like the idea and think it could be very useful 👍

**Comment by KPull** [comment]
I will be working on this issue unless anyone has any further suggestions or comments to make.

**Comment by KPull** [comment]
Implemented in #88.
